app/upgrades/v9/upgrades.go

```diff
@@ -0,0 +1,65 @@
+package v9
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	distrKeeper "github.com/cosmos/cosmos-sdk/x/distribution/keeper"
+	upgradetypes "github.com/cosmos/cosmos-sdk/x/upgrade/types"
+	"github.com/evmos/evmos/v9/types"
+)
+
+// CreateUpgradeHandler creates an SDK upgrade handler for v9
+func CreateUpgradeHandler(
+	mm *module.Manager,
+	configurator module.Configurator,
+	dk distrKeeper.Keeper,
+) upgradetypes.UpgradeHandler {
+	return func(ctx sdk.Context, _ upgradetypes.Plan, vm module.VersionMap) (module.VersionMap, error) {
+		logger := ctx.Logger().With("upgrade", UpgradeName)
+
+		if types.IsMainnet(ctx.ChainID()) {
+			logger.Debug("recovering lost funds from clawback...")
+			if err := ReturnFundsFromCommunityPool(ctx, dk); err != nil {
+				// log error instead of aborting the upgrade
+				logger.Error("FAILED TO RECOVER FROM COMMUNITY FUNDS", "error", err.Error())
+			}
+		}
+
+		// Leave modules are as-is to avoid running InitGenesis.
+		logger.Debug("running module migrations ...")
+		return mm.RunMigrations(ctx, configurator, vm)
+	}
+}
+
+func ReturnFundsFromCommunityPool(ctx sdk.Context, dk distrKeeper.Keeper) error {
+	availableCoins, ok := sdk.NewIntFromString(MaxRecover)
+	if !ok {
+		return fmt.Errorf("failed to read amount to recover from community funds")
+	}
+	for i := range Accounts {
+		refund, _ := sdk.NewIntFromString(Accounts[i][1])
+		if availableCoins.LT(refund) {
```

Comment:
we should also check that `availableCoins` is never negative

Reply:
that would mean that `LT` and `Sub` are failing, and we should panic in that case.

I added the check to see if the initial value is not negative.
We can also check if every value in account is valid.